First version of SITCOM-1593 notebook, ready for review

nsevilla opened the pull request.

nsevilla committed:
First version of SITCOM-1593 notebook, ready for review

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    %load_ext lab_black
</pre><p>Hi Nacho - Unfortunately, the DM folks are removing lab_black from RSP. So some people might have problems running the notebook with this. Can you remove it, please?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135689'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

nsevilla replied:
OK!

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<p>Instead of <code>EfdClient</code> , I recommend you use <code>makeEfdClient</code> . This will create an efd client independently if you are running this notebook at the summit or at USDF.   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135692'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

nsevilla replied:
OK!

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #3.    n_actuators = 156
</pre><p>This is a constant. I would move it to the same cell together with the imports with all letters capital <code>N_ACTUATORS</code> . </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135695'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

nsevilla replied:
OK!

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #4.    primary_FA_error = [
</pre><p>If might be more compact to write something like: </p><p><code>primary_FA_error = [f"primaryCylinderFollowingError{i}" for i in range(n_actuators)"]</code> </p><p>You can even consider adding the <code>[...]</code> directly to the query to get data from the Efd.   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135696'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

nsevilla replied:
Used your proposed syntax.

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #7.    df_primary_FA_error = await client.select_time_series(
</pre><p>Instead of using <code>await client.select_time_series</code> consider using <code>getEfdData</code> which you have imported. The function does basically the same without needing the <code>await</code>.    </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135699'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

nsevilla replied:
Indeed, I think I copy-pasted something very old because many of the options here I have been using in my more recent notebooks.

nsevilla replied:
<p>Updated</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<p>Is there any especial reason to keep this code? If this is not being used anymore and you are not planning to use it, maybe remove it?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135701'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

nsevilla replied:
<p>Removed. </p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    resample_in_sec = 60  # seconds
</pre><p>I would move this up in the notebook since it seems that it is a parameter that is used in several places. In this case, I would keep the name of the variable as it is. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135702'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

nsevilla replied:
<p>moved</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #2.    # take the mean value for each of the actuators in {resample_in_sec} second samples
</pre><p>I would add a empty line between every comment line in this cell. This give a bit more of space in the code and improves readability.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135703'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #4.    # get the actuator ID which has a maximum or minimum value in the resampled data set
</pre><p>Add a space here. Same reason as above.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135704'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='12'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #2.    n_bins = 30
</pre><p>I would move this variable to the beginning of the notebook. Maybe have a single cell with the variables that the user can change? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135705'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='14'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #8.    plt.scatter(
</pre><p>Add space here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135708'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='14'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #16.    plt.scatter(
</pre><p>add space here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135709'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='14'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #24.    plt.scatter(
</pre><p>add space here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135711'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='14'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #7.    plt.xlabel("Actuator ID")
</pre><p>Add space here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135713'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='16'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    n_bins = 30
</pre><p>See that <code>n_bins</code> is repeated here. I would delete this one once the other is setup earlier in the notebook.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135715'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='18'/>

nsevilla replied:
OK!

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #9.    for j in range(n_actuators):
</pre><p>Add space before this line</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135716'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='18'/>

nsevilla replied:
OK!

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #28.    bin_centers = (bin_edges[:-1] + bin_edges[1:]) / 2
</pre><p>Add space before this line.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135717'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='18'/>

nsevilla replied:
OK!

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    from collections import defaultdict
</pre><p>I would move all these imports to the import cell in the beginning of the notebook.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135719'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

nsevilla replied:
OK!

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #25.        fadata_min["FA"].append(f"Force Actuator {i}")
</pre><p>I recommend grouping the code here. Something like:</p>

<pre class="ql-syntax" spellcheck="false">for ...
    (optional empty line)
&nbsp; &nbsp; fadata_mean["elevation"].append(bin_centers)
&nbsp; &nbsp; fadata_mean["force error"].append(fa_mean[i])
&nbsp; &nbsp; fadata_mean["FA"].append(f"Force Actuator {i}")
&nbsp; &nbsp; 
    fadata_max["elevation"].append(bin_centers)
&nbsp; &nbsp; fadata_max["force error"].append(fa_max[i])
&nbsp; &nbsp; fadata_max["FA"].append(f"Force Actuator {i}")
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859135721'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

nsevilla replied:
<p>Sorry, I don't quite understand this comment. How is this any different to what is there already? There is a for loop filling in all these dictionaries as you suggest.</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #6.    max_actuators = np.argmax(df_primary_FA_error_resampled_max, axis=1)
</pre><p>It seems that this is only used in a plot a few cells below. Maybe bring line 6 and 7 closer to their associated plot so it is easier to realize where the data is coming from?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859151890'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='12'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<p>I wonder if having semilog plot would make it easier to visualize... Maybe add horizontal grid lines?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859151893'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='16'/>

nsevilla replied:
<p>OK!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<p>This plot looks awesome! It is exactly what I was thinking. We can see quickly how FA 67 is different from the others.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859151895'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

nsevilla replied:
<p>Great!</p>

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1593_force_actuator_error_elevation.ipynb:
```diff
@@ -0,0 +1,514 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #16.    for i in range(n_actuators):
</pre><p>The only detail to change here would be the parameter you are using to identify the force actuators. Right now, you are using the index. It would be better to have the actuator id. If you print the <code>FATable</code> variable, you will see something like:</p>

<pre class="ql-syntax" spellcheck="false">ForceActuatorData(index=42, actuator_id=143, x_position=1.60401001, y_position=-3.692780029, z_position=-2.158743, actuator_type=&lt;FAType.SAA: 0&gt;, subnet=&lt;FASubnet.C: 3&gt;, address=3, orientation=&lt;FAOrientation.NA: 0&gt;, x_index=None, y_index=None, z_index=42, s_index=None, near_neighbors=[140, 139, 134, 133, 138, 142], far_neighbors=[139, 140, 142, 138, 134, 133, 135, 141, 128, 137, 132, 129]), 
</pre><p>The <code>actuator_id</code> is what we should display.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/148/discussion/#comment-1859151897'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

nsevilla replied:
<p>Indeed! Thanks for catching this, I didn't know of this subtle detail. In fact, it led me to modifying the histogram before it as well.</p>

nsevilla committed:
Addressing PRs comments

b1quint commented:
Hi Nacho, could you squash your last commits with the previous one(s)? I ask because the outputs remain in the history line forever if you just keep another commit, and they increase the size of the repository.

nsevilla commented:
> Hi Nacho, could you squash your last commits with the previous one(s)? I ask because the outputs remain in the history line forever if you just keep another commit, and they increase the size of the repository.

@b1quint can you check if it is fine now?

nsevilla committed:
Added histograms with hovering information

Added histograms with hovering information, fixed static histogram bins. Removed unused block and added histogram title.

b1quint reviewed (approved):
Alright, all the branch issues seem solved. Thank you for that! Approved! 👍🏽
